### PR TITLE
First working example with a queue

### PR DESCRIPTION
**Pull request by caggle**
In this commit, we create an SQS queue via serverless.yml file, and 2 more functions.

- One function reads list of hosts from the same json list used for scheduled scans, and puts them in the queue

- The second function simply prints out the next message in queue (event). That's it for now.

Note that this is a prototype.

### CODE REVIEW

**Comment by caggle** [comment]
@claudijd Can you please review this?

It uses an SQS queue in a very simple way. Has 3 functions:
1. `ingest` function that is invoked manually, which picks a host from a pre-defined list and adds to the queue.
2. `ObservatoryScanQueue` function runs as soon as there is a new entry in the queue. Currently this does not run a scan, simply prints the hostname to the logs as an example.
3. `cronObservatoryScan` function picks a host from a pre-defined list and runs Observatory scan on it every 5 minutes.

If all is well I will build on top of this (perform port scans and add CT logs as ingestion point).

File: handler.py [hunk 1]
```diff
@@ -1,33 +1,68 @@
 import json
-import datetime
 import logging
+import boto3
+import os
 
 from util.s3_helper import send_to_s3
+from util.validator import validateTarget
```
Comment (claudijd) [inline handler.py]: Consider renaming "validTarget" to "TargetValidator", it moves from an action to a noun, which I believe is more OOP proper.

File: util/validator.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+from lib.target import Target
+
+
+def validateTarget(host):
```
Comment (claudijd) [inline util/validator.py]: We already have most of this functionality added to the Target object, what does making this another object serve?

File: handler.py (hunk 1 above)
Comment (claudijd) [reply handler.py]: Scratch that, I think we should just make use of the existing Target object

File: handler.py [hunk 3]
```diff
@@ -1,33 +1,68 @@
...
+def putInQueue(event, context):
+    # For demo purposes, this function is invoked manually
+    # Also for demo purposes, we will use a static list here
+    # We need to figure out a way to put stuff in the queue regularly
+    target_list = [
+        "www.mozilla.org",
+        "infosec.mozilla.org",
+    ]
+    randomizer = Randomizer(target_list)
+    target = Target(randomizer.next())
```
Comment (claudijd) [inline handler.py]: I think the validity checking should be done here, before we put it into the queue.  By the time we run a scan, we should have confidence it's valid, but if we want to be paranoid, we could check in both places.

File: handler.py [hunk 4]
```diff
@@ -1,33 +1,68 @@
...
+
+    # This is how we process the hostname
+    # Demonstration purposes, in the final example 
+    # we should extract the hostname and perform a scan
+
+    for record, keys in event.items():
+        for item in keys:
+            if "body" in item:
+                logger.info(item['body'])
+    return ''
```
Comment (claudijd) [inline handler.py]: What's the purpose of returning an empty string always on this method?

File: handler.py [hunk 5]
```diff
@@ -1,33 +1,68 @@
...
+        "infosec.mozilla.org",
+    ]
+    randomizer = Randomizer(target_list)
+    target = Target(randomizer.next())
+
+    print(SQS_CLIENT.send_message(
+        QueueUrl=os.getenv('SQS_URL'),
+        MessageBody=target.targetname
+    ))
+    return ''
```
Comment (claudijd) [inline handler.py]: Same here, what purpose does returning  an empty string serve?

File: handler.py (hunk 1 above)
Comment (caggle) [reply handler.py]: No worries, happy to discard this. I only created this function to reduce repetition in `handler.py` as eventually there will be multiple ingestion points which will consume a hostname or an IP address.

File: util/validator.py (hunk 2 above)
Comment (caggle) [reply util/validator.py]: Will remove.

File: handler.py (hunk 3 above)
Comment (caggle) [reply handler.py]: Agree.

File: handler.py (hunk 4 above)
Comment (caggle) [reply handler.py]: Copy/paste error :) will remove.

File: handler.py (hunk 5 above)
Comment (caggle) [reply handler.py]: Again, copy paste error, my bad.

**Pull request merged**